### ISSUE TITLE
Invalid selector can cause match() to hang

### ISSUE THREAD

**Issue by domenic** [dperini/nwmatcher#97]
When using the selector `//MAIN MENU - (used to keep mobile menu options hidden and keep weather/search and menu on one line) // #tncms-region-nav-main-nav-right-nav`, I have made nwmatcher hang.

This is the ultimate cause of https://github.com/tmpvar/jsdom/issues/1214. (Our CSS parser engine interprets the crazy stylesheet as having that as the selector, which then triggers this bug. It probably shouldn't be interpreting that as a selector though.)

**Comment by dperini** [comment]
I am currently in Ecuador until 16 September and unable to look at this bug.
Could you please write a small Javascript example showing the problem ?
"//MAIN MENU" is not a selector, it is strange that it passes the validator !

**Comment by domenic** [comment]
I added such a test using jsdom in https://github.com/tmpvar/jsdom/commit/7c0bf4a88e7e9f1acf4508d6d11d404f17892629#diff-27b29d939099042e6e36b98eb29abd15R25:

```js
  const document = jsdom();

  const selector = " //MAIN MENU - (used to keep mobile menu options hidden and keep weather/search and menu " +
                   "on one line) // #tncms-region-nav-main-nav-right-nav";
  
  // next line will hang forever
  document.querySelector(selector);
```

If you'd like an example that only uses nwmatcher let me know and I can work on that too.

**Comment by dperini** [comment]
Domenic,
I believe I have found a simple fix for this, I am just a bit short on time.
I will probably be able to work on it this weekend.

**Comment by domenic** [comment]
Awesome, thank you!

**Comment by dperini** [comment]
Fixed in 3595c7d... sorry for the delay.
Please test so I can make a new release and publish to npm !

**Comment by domenic** [comment]
@dperini tested, that fixes it! Thank you! Please publish whenever you can. Sorry for the delay on my end.

**Comment by dperini** [comment]
Published 1.3.7 ... me too part of the delay  :)
